code/datums/quirks/negative.dm

```diff
@@ -424,33 +424,63 @@
 
 /datum/quirk/prosthetic_limb
 	name = "Prosthetic Limb"
-	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a random prosthetic!"
+	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a prosthetic replacement!"
 	icon = "tg-prosthetic-leg"
 	value = -3
 	var/slot_string = "limb"
 	medical_record_text = "During physical examination, patient was found to have a prosthetic limb."
 	hardcore_value = 3
+	var/prosthetic_choice
 
+
+// i've made a real fuckin mess of this frankenstein ass bullshit code i have no idea what i'm doing
 /datum/quirk/prosthetic_limb/add_unique()
-	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	prosthetic_choice = prosthetic_choice || quirk_holder.client?.prefs?.read_preference(/datum/preference/choiced/prosthetic)
```

Comment:
Not sure what `prosthetic_choice = prosthetic_choice || ...` is doing here that is useful, you probably just want
`prosthetic_choice = quirk_holder.client?.prefs?.read_preference(/datum/preference/choiced/prosthetic)`

Reply:
I had more or less copied that bit from the phobia quirk, wasn't sure if the || part was necessary

Reply:
It _might_ be I just don't know why 😅

Reply:
If prosthetic_choice choice is null, read from prefs?